raiden/tests/property/smart_contracts/test_nettingchannel.py

```diff
@@ -0,0 +1,333 @@
+            pkey_address[1],
+            sender=pkey_address[0],
+        )
+
+        if not self.is_participant(pkey_address):
+            try:
+                self.netting_channel.deposit(  # pylint: disable=no-member
+                    deposit_amount,
+                    sender=pkey_address[0],
+                )
+            except TransactionFailed:
+                pass
+            else:
+                raise ValueError('deposit from non-participant didnt fail')
+
+        elif self.netting_channel.closed(sender=pkey_address[0]) != 0:  # pylint: disable=no-member
+            try:
+                self.netting_channel.deposit(  # pylint: disable=no-member
+                    deposit_amount,
+                    sender=pkey_address[0],
+                )
+            except TransactionFailed:
+                pass
+            else:
+                raise ValueError('deposit with closed channel didnt fail')
+
+        elif token_balance < deposit_amount:
+            try:
+                self.netting_channel.deposit(  # pylint: disable=no-member
+                    deposit_amount,
+                    sender=pkey_address[0],
+                )
+            except TransactionFailed:
+                pass
+            else:
+                raise ValueError('deposit without balance didnt fail')
+
+        else:
+            self.netting_channel.deposit(  # pylint: disable=no-member
+                deposit_amount,
+                sender=pkey_address[0],
+            )
+
+    def contract_close(self, transfer, pkey_address):
+        transfer_data = transfer.encode()
+
+        if not self.is_participant(transfer.sender):
+            try:
+                self.netting_channel.close(  # pylint: disable=no-member
+                    transfer_data,
+                    sender=pkey_address[0],
+                )
+            except TransactionFailed:
+                pass
+            else:
+                raise ValueError('close with transfer data from a non participant didnt fail')
+
+        elif transfer.sender == pkey_address[1]:
+            try:
+                self.netting_channel.close(  # pylint: disable=no-member
+                    transfer_data,
+                    sender=pkey_address[0],
+                )
+            except TransactionFailed:
+                pass
+            else:
+                raise ValueError('close with self signed transfer didnt fail')
+
+        elif self.netting_channel.closed(sender=pkey_address[0]) != 0:  # pylint: disable=no-member
+            try:
+                self.netting_channel.close(  # pylint: disable=no-member
+                    transfer_data,
+                    sender=pkey_address[0],
+                )
+            except TransactionFailed:
+                pass
+            else:
+                raise ValueError('close called twice didnt fail')
+
+        elif self.is_participant(pkey_address[1]):
```

Comment:
Wait, you're correct, this should be a `not self.is_participant(pkey_address[1]):`, it's handling the third key `sha3('p3')`